src/systems/filecoin_nodes/network/_index.md

```diff
@@ -31,6 +31,24 @@ Here is the list of libp2p protocols used by Filecoin.
 - DNSDiscovery: Design and spec needed before implementing
 - HTTPDiscovery: Design and spec needed before implementing
 - Hello:
-	- Hello protocol handles new connections to filecoin nodes.  It is an important part of the discovery process for ambient protocols (like KademliaDHT)
-	- Spec TODO.
-	- No Filecoin specific modifications to the protocol id.
+	- Hello protocol handles new connections to filecoin nodes to facilitate discovery
+	- the protocol string is `fil/hello/1.0.0`. 
```

Comment:
I think this protocol string should include the network name, as the pubsub topics do.

Reply:
@whyrusleeping @anorth you two appear to disagree.  I can't merge this until I hear both sides of the story, please express your viewpoints in the comments below.  Thanks!

Reply:
So, if we don't have the hello protocol on a common protocol name, nodes on different networks won't discover they are on different networks. Lotus triggers a disconnect if it connects to a node on a different genesis block

Reply:
This is a good point.  Rephrasing: as implemented by lotus, a node will leave open connections that do not respond to hellos but will close connections that respond to hellos with different genesis cids (and different networks always have different genesis cids).  If we separate out different networks at the protocol negotiation layer lotus would actually keep open the connections to nodes that are on a different network because they'd never even run hello.

Reply:
Ok, so they _could_ discover it by inspecting the protocols, or just learning that a peer doesn't implement the expected protocol. I gather Lotus desires to keep open connections that don't implement hello at all?

It seems like all behaviours could be implemented in either case, so I don't mind. No network name is ok.

Reply:
> I gather Lotus desires to keep open connections that don't implement hello at all?

Yeah I think that is really the crux of this design decision if you pull on it long enough.  Its definitely something @whyrusleeping believes in strongly for interop across many different protocols.  Glad you are calling it out explicitly because it is not written down anywhere.